lua/cmp/entry.lua

```diff
@@ -223,6 +224,7 @@ entry.get_view = function(self, suggest_offset, entries_buf)
       view.menu.text = item.menu or ''
       view.menu.bytes = #view.menu.text
       view.menu.width = vim.fn.strdisplaywidth(view.menu.text)
+      view.menu.hl_group = has_border and 'CmpBorderedWindow' or 'CmpWindow'
```

Comment:
This is not related to border/window! You don't need changing this in this PR.

Reply:
I just want to make sure I understand correctly:

* `CmpItemMenu` is the color of the completion menu (a popup window).
* `CmpWindow` and `CmpBorderedWindow` are new highlight groups to be used for coloring popup windows with and without borders.

So, when the custom completion menu has a border, we need to respect the user's `CmpWindow` or `CmpBorderedWindow` setting, right?

Because, when I color `CmpItemMenu` on the main branch, it affects the color of the completion menu popup like the `CmpWindow` or `CmpBorderedWindow` would be expected to. Are these new highlight groups just for the documentation windows?

Sorry for all the questions; it seems like these three highlight groups are doing the same thing. I might be misunderstanding, or we should unify them.

Reply:
- The `CmpItemMenu` is for highlighting the `menu field` color.
    - https://github.com/hrsh7th/nvim-cmp/blob/main/lua/cmp/view/custom_entries_view.lua#L76
- The `completion menu` color is `Pmenu` by default.
  - https://github.com/hrsh7th/nvim-cmp/blob/main/lua/cmp/view/custom_entries_view.lua#L35

The `CmpWindow` and `CmpBorderedWindow` should be set up via `winhighlight` option.
Currently, nvim-cmp set up it once at initialization. But after this PR, the `winhighlight` option will be depending on to configuration setting so you should move it more suitable location.

Reply:
Ah. I think I see where we're misunderstanding each other.

I was thinking that since a completion menu is filled with `menu field`s, the `CmpWindow` color will be hidden behind `CmpItemMenu`. 

So I guess we're both _technically_ right:

* `CmpItemMenu` is **not** a popup color, like you say.
* `CmpItemMenu` **acts** like a popup color because it hides `CmpWindow`/`CmpBorderedWindow`, like I was thinking.

The question is: how should we fix this? Maybe set `CmpItemMenu`'s default based on whether there is a border?

Reply:
No. the completion menu contains `abbr field` , `kind field` and `menu field`.